src/axis/Axis.js

```diff
@@ -555,6 +568,129 @@ export default class Axis {
 		return currentTickMax.size;
 	}
 
+	getXAxisTickTextY2Overflow(defaultPadding) {
+		const $$ = this.owner;
+		const config = $$.config;
+		const xAxisTickRotate = $$.getAxisTickRotate("x");
+		const positiveRotation = xAxisTickRotate > 0 && xAxisTickRotate < 90;
+		const isTimeSeries = $$.isTimeSeries();
+		const isCategorized = $$.isCategorized();
+		const allowedAxisTypes = isCategorized || isTimeSeries;
+
+		if (
+			$$.svg &&
+			allowedAxisTypes &&
+			!config.axis_x_tick_culling &&
+			!config.axis_x_tick_multiline &&
+			positiveRotation
+		) {
+			const widthWithoutCurrentPaddingLeft = $$.currentWidth - $$.getCurrentPaddingLeft();
+			const maxOverflow = this.getMaxOverflow(xAxisTickRotate, widthWithoutCurrentPaddingLeft);
```

Comment:
This leads follow stack calls:
- `this.getMaxOverflow()` -> `this.getXAxisPadding()` -> `this.calculateXAxisPadding()`.

They're returning same values whenever resize happens in most cases.
Could you make only called when they're necessary, to reduce the cost?

Reply:
During the refactoring I came across some problems with `axis_x_tick_fit: false` which I have overlooked.

As described [here](https://naver.github.io/billboard.js/release/latest/doc/Options.html#.axis%25E2%2580%25A4x%25E2%2580%25A4tick%25E2%2580%25A4fit), using `axis_x_tick_fit: true` we have ticks with same intervals, which is good for our calculations.
As soon as we use `axis_x_tick_fit: false`, the intervals can be different, as they are calculated by d3.
As the reason for not supporting `axis_x_type: "indexed"` were intervals with different lengths, I came to the conclusion that supporting `axis_x_tick_fit: false` makes no sense.

I already removed unnecessary calls of `this.getMaxOverflow() -> this.getXAxisPadding() -> this.calculateXAxisPadding()` in my local branch and did some other refactorings.

If supporting only `axis_x_tick_fit: true` is ok for you, I will push my changes asap.

Reply:
@michkami, well is okay for now. But as of `autorotate` works only in certain conditions, It needs to clarify more in detail from API doc.
> Rotate x axis tick text if there is not enough space.

like as:
> Rotate x axis tick text if there is not enough space for 'category' and 'timeseries' type axis with same intervals tick texts.

BTW, adding new feature is always challenging, because it needs to consider the compatibility with others options to not break or having unexpected behavior combining them.